src/Microsoft.ComponentDetection.Detectors/nuget/NuGetNuspecUtilities.cs

```diff
@@ -45,20 +91,68 @@ public static async Task<byte[]> GetNuspecBytesAsync(Stream nupkgStream)
             finally
             {
                 // make sure that no matter what we put the stream back to the beginning
-                nupkgStream.Seek(0, SeekOrigin.Begin);
+                nupkgComponentStream.Stream.Seek(0, SeekOrigin.Begin);
+            }
+        }
+
+        public static (string Name, string Version, string[] Authors, HashSet<string> TargetFrameworks) GetNuspecDataFromNuspecStream(Stream sourceStream)
```

Comment:
There is a question of whether we should assume a directory containing a .nuspec file is an expanded .nupkg. There may be adjacent directories and subdirectories that could contain relevant target framework data that perhaps should be analyzed as in GetNuGetPackageDataFromNupkg above.